src/Microsoft.DotNet.ProjectJsonMigration/PackageConstants.cs

```diff
@@ -26,55 +26,1810 @@ internal class PackageConstants
         public const string DotnetTestXunit = "dotnet-test-xunit";
         public const string DotnetTestMSTest = "dotnet-test-mstest";
 
-        public static readonly IDictionary<string, PackageDependencyInfo> ProjectDependencyPackages = 
-            new Dictionary<string, PackageDependencyInfo> {
-                {"Microsoft.EntityFrameworkCore.Tools", new PackageDependencyInfo {
-                    Name = "Microsoft.EntityFrameworkCore.Tools",
-                    Version = ConstantPackageVersions.AspNetToolsVersion } },
-                { "Microsoft.AspNetCore.Razor.Tools", null },
-                { "Microsoft.AspNetCore.Razor.Design", null },
-                { "Microsoft.VisualStudio.Web.CodeGenerators.Mvc", new PackageDependencyInfo {
-                    Name = "Microsoft.VisualStudio.Web.CodeGeneration.Design",
-                    Version = ConstantPackageVersions.AspNetToolsVersion } },
-                { "Microsoft.VisualStudio.Web.CodeGeneration.Tools", null},
-                { TestSdkPackageName, new PackageDependencyInfo {
-                    Name = TestSdkPackageName,
-                    Version = ConstantPackageVersions.TestSdkPackageVersion } },
-                { XUnitPackageName, new PackageDependencyInfo {
-                    Name = XUnitPackageName,
-                    Version = ConstantPackageVersions.XUnitPackageVersion } },
-                { XUnitRunnerPackageName, new PackageDependencyInfo {
-                    Name = XUnitRunnerPackageName,
-                    Version = ConstantPackageVersions.XUnitRunnerPackageVersion } },
-                { MstestTestAdapterName, new PackageDependencyInfo {
-                    Name = MstestTestAdapterName,
-                    Version = ConstantPackageVersions.MstestTestAdapterVersion } },
-                { MstestTestFrameworkName, new PackageDependencyInfo {
-                    Name = MstestTestFrameworkName,
-                    Version = ConstantPackageVersions.MstestTestFrameworkVersion } },
-                { DotnetTestXunit, null },
-                { DotnetTestMSTest, null },
+        public static readonly IDictionary<PackageDependencyInfo, PackageDependencyInfo> ProjectDependencyPackages =
+            new Dictionary<PackageDependencyInfo, PackageDependencyInfo> {
+                {
+                    new PackageDependencyInfo {
+                        Name = "Microsoft.EntityFrameworkCore.Tools",
```

Comment:
Agreed, but I think having it in C# is less fragile, and with a little refactoring it can be just as terse in C# as, for example, in JSON, without adding ways to fail due to malformed JSON, bad file locations, etc. Configuration is still code. Reducing the number of tools engaged is also a maintainability win.